crates/store/src/state_db.rs

```diff
@@ -168,6 +168,24 @@ impl CheckPoint {
         db: &StoreTransaction,
         db_mode: StateDBMode,
     ) -> Result<(u64, u32), Error> {
+        let block_offset = |withdrawal_count: u32, tx_count: u32| -> u32 {
+            if 0 == withdrawal_count {
```

Comment:
We can replace the `if` statement with
`(withdrawal_count + 1 + tx_count).saturating_sub(1)`